### PR TITLE
PIM-9754: Able the STD edition upgrade to fix Akeneo managed files

### PR DESCRIPTION
**Pull request by mr-prud**
**Description (for Contributor and Core Developer)**

During a standard minor update on v5, some files are no updated, to avoid manual operation we move and rework update process and configuration loading behavior

**Definition Of Done (for Core Developer only)**

Allow comments in env var (allow by symfony) on CI
https://symfony.com/doc/4.4/configuration.html#configuring-environment-variables-in-env-files

## Upgrade Process

We must upgrade some files during the composer upgrade.

* Doctrine migration scripts.
* Akeneo managed files ( dot env and kernel.php).

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

**Review by greg-akeneo — APPROVED** [review]

File: .env [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# DON'T EDIT THIS FILE !
```
Comment (BitOne) [inline .env:1]: To remove... I think it's going to break the application...

File: .circleci/detect_structure_changes.sh [hunk 2]
```diff
@@ -87,7 +87,7 @@ echo "Copy CE migrations into EE to launch branch migrations..."
 cp -R vendor/akeneo/pim-community-dev/upgrades/schema/* upgrades/schema
 
 echo "Export env vars from .env..."
-export $(cat .env)
+export $(cat .env | grep -v "^#")
```
Comment (BitOne) [inline .circleci/detect_structure_changes.sh:90]: Don't put comment inside .env files :)

**Review by BitOne — APPROVED** [review]
GTM after fixes

File: .env (hunk 1 above)
Comment (mr-prud) [reply .env:1]: Like I said in the PR description, comment are allowed in the dotenv symfony module
https://symfony.com/doc/4.4/configuration.html#configuring-environment-variables-in-env-files

File: std-build/install-required-files.sh [hunk 3]
```diff
@@ -11,13 +11,30 @@ set -e
 DEV_DISTRIB_DIR=$(dirname $0)/..
 STANDARD_DISTRIB_DIR=./
 
+# Step: Install and upgrade
+###########################
+
+# Update doctrine migratation files
```
Comment (StevenVAIDIE) [inline std-build/install-required-files.sh:17]: :smile: 
```suggestion
# Update doctrine migration files
```

**Review by StevenVAIDIE — APPROVED** [review]

File: .circleci/detect_structure_changes.sh (hunk 2 above)
Comment (mr-prud) [reply .circleci/detect_structure_changes.sh:90]: Comments only break the CI

File: .env (hunk 1 above)
Comment (mr-prud) [reply .env:1]: And the application works with this file.

**Pull request merged**